tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Logging/NLog/DirectSubmission/DirectSubmissionNLogTarget.cs

```diff
@@ -0,0 +1,88 @@
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.Reflection;
+using Datadog.Trace.ClrProfiler.AutoInstrumentation.Logging.NLog.DirectSubmission.Formatting;
+using Datadog.Trace.ClrProfiler.AutoInstrumentation.Logging.NLog.DirectSubmission.Proxies;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Logging.DirectSubmission;
+using Datadog.Trace.Logging.DirectSubmission.Formatting;
+using Datadog.Trace.Logging.DirectSubmission.Sink;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Logging.NLog.DirectSubmission
+{
+    /// <summary>
+    /// NLog Target that sends logs directly to Datadog
+    /// </summary>
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class DirectSubmissionNLogTarget
+    {
+        private readonly IDatadogSink _sink;
+        private readonly int? _minimumLevel;
+        private readonly LogFormatter _formatter;
+        private TargetWithContextBaseProxy _baseProxy;
+
+        internal DirectSubmissionNLogTarget()
+        {
+        }
+
+        // internal for testing
+        internal DirectSubmissionNLogTarget(
+            IDatadogSink sink,
+            DirectSubmissionLogLevel? minimumLevel,
+            LogFormatter formatter)
+        {
+            _sink = sink;
+            _formatter = formatter;
+            _minimumLevel = (int?)minimumLevel;
+        }
+
+        /// <summary>
+        /// Writes logging event to the log target
+        /// </summary>
+        /// <param name="logEventInfo">Logging event to be written out</param>
+        [DuckReverseMethod(ParameterTypeNames = new[] { "NLog.LogEventInfo, NLog" })]
+        public void Write(LogEventInfoProxy logEventInfo)
+        {
+            if (logEventInfo is null)
+            {
+                return;
+            }
+
+            if (logEventInfo.Level.Ordinal < (_minimumLevel ?? (int)TracerManager.Instance.DirectLogSubmission.Settings.MinimumLevel))
+            {
+                return;
+            }
+
+            // Nlog automatically includes all the properties from the event, so don't need fallback properties
+            var mappedProperties = _baseProxy.GetAllProperties(logEventInfo);
+
+            // We render the event to a string immediately as we need to capture the properties
+            // This is more expensive from a CPU perspective, but is necessary as the properties
+            // won't necessarily be serialized correctly otherwise (e.g. dd_span_id/dd_trace_id)
+
+            var logEvent = new LogEntry(logEventInfo, mappedProperties, fallbackProperties: null);
+            var logFormatter = _formatter ?? TracerManager.Instance.DirectLogSubmission.Formatter;
+            var serializedLog = NLogLogFormatter.FormatLogEvent(logFormatter, logEvent);
```

Comment:
This applies to all the other targets as well, but if the sink is the null datadog sink (i.e., direct submission is not enabled) do we want to format the log at all?

Reply:
If we're using the null sink, we won't get this far, as direct log shipping will be disabled